### DISCUSSION TITLE
Measurement Tooltip

### DISCUSSION THREAD

**Discussion by dvazquez-lightbox** [geoman-io/leaflet-geoman#1529] [Questions & Help] [answered]
Hi, I was wondering if it was possible to modify the tooltip content to display data other than those defined under measurements options. I can see a method: addMeasurementTooltipToLayer but there is not much information available for the feature in the docs.

**Comment by Falke-Design** [comment, marked as answer]
Hi @dvazquez-lightbox,
there is no method to add own content to the tooltip but you can overwrite it like that:
```
map.on("pm:drawstart", ({ workingLayer, shape }) => {
    if(shape === 'Rectangle'){
      workingLayer.on("pm:change", () => {
          map.pm.Draw.Rectangle._hintMarker.getTooltip().setContent(JSON.stringify(workingLayer.pm.measurements));
      });
    }
});
```

**Reply by dvazquez-lightbox** [reply]
Thank you, this worked perfectly!

**Reply by dvazquez-lightbox** [reply]
@Falke-Design In a similar vein, is there a way to set the tooltip on pm:drawend i know _hintmarker does it when drawing actively, but is there something similar when the user is not currently drawing but has finished.

**Reply by Falke-Design** [reply]
You can just simple create your own tooltip: 

```
const tooltip = L.tooltip({
          permanent: true,
          offset: L.point(0, 10),
          direction: 'bottom',
          opacity: 0.8,
        })
    .setLatLng(latlng)
    .setContent('Hello world!<br />This is a nice tooltip.')
    .addTo(map);
```
Docs: https://leafletjs.com/reference.html#tooltip

**Reply by dvazquez-lightbox** [reply]
awesome, was just wondering if there was any internal method inside like _hintmarker, but this works too. thank you
